client/src/components/Button.js

```diff
@@ -0,0 +1,70 @@
+import styled from '@emotion/styled';
+
+const Button = styled.button`
+  cursor: pointer;
+  font-family: 'Quicksand Book', sans-serif;
+  padding: ${(props) => buttonSize[props.size].padding};
+  font-size: ${(props) => buttonSize[props.size].fontsize};
+  width: ${(props) => buttonSize[props.size].width};
+  text-transform: uppercase;
+  color: #000;
+  background: #f9f9f9;
+  background-clip: padding-box;
+  border: none;
+  border-radius: 1rem;
+  position: relative;
+
+  &:before {
+    content: '';
+    display: block;
+    position: absolute;
+    width: ${(props) => buttonSize[props.size].borderWidth};
+    height: ${(props) => buttonSize[props.size].height};
+    top: 50%;
+    transform: translateY(-50%);
+    left: -0.5%;
+    z-index: -1;
+    margin: 0;
+    border-radius: 1rem;
+    background: linear-gradient(
+      to right,
+      #fff700,
+      #ff9100,
+      #ff0084,
+      #ff00f2,
+      #bb00ff,
+      #7700ff,
+      #006aff,
+      #00ccff,
+      #00ff6a,
+      #59ff00,
+      #fff700
+    );
```

Comment:
maybe this helps with compatibility. A different approach with just borders and no extra elements:

- the `9001px` is just a joke, it needs to be half the height ;)
- `linear-gradient(white, white)` is the inner part of the button

```suggestion
  height: ${(props) => buttonSize[props.size].height};
  border: double 4px transparent;
  border-radius: 9001px;
  background-image: linear-gradient(white, white), linear-gradient(to right,
      #fff700,
      #ff9100,
      #ff0084,
      #ff00f2,
      #bb00ff,
      #7700ff,
      #006aff,
      #00ccff,
      #00ff6a,
      #59ff00,
      #fff700);
   background-origin: border-box;
   background-clip: padding-box, border-box;
```

Reply:
this worked so much better!